avm/res/compute/virtual-machine/main.bicep

```diff
@@ -618,19 +820,125 @@ module vm_azureDiskEncryptionExtension 'extension/main.bicep' = if (extensionAzu
+    typeHandlerVersion: contains(extensionAzureDiskEncryptionConfig, 'typeHandlerVersion') 
+     ? extensionAzureDiskEncryptionConfig.typeHandlerVersion 
+     : (osType == 'Windows' ? '2.2' : '1.1')
+    autoUpgradeMinorVersion: contains(extensionAzureDiskEncryptionConfig, 'autoUpgradeMinorVersion') 
+     ? extensionAzureDiskEncryptionConfig.autoUpgradeMinorVersion 
+     : true
+    enableAutomaticUpgrade: contains(extensionAzureDiskEncryptionConfig, 'enableAutomaticUpgrade') 
+     ? extensionAzureDiskEncryptionConfig.enableAutomaticUpgrade 
+     : false
+    forceUpdateTag: contains(extensionAzureDiskEncryptionConfig, 'forceUpdateTag') 
+     ? extensionAzureDiskEncryptionConfig.forceUpdateTag 
+     : '1.0'
     settings: extensionAzureDiskEncryptionConfig.settings
     tags: extensionAzureDiskEncryptionConfig.?tags ?? tags
   }
   dependsOn: [
     vm_customScriptExtension
-    vm_azureMonitorAgentExtension
   ]
 }
 
+module vm_nvidiaGpuDriverWindowsExtension 'extension/main.bicep' = if (extensionNvidiaGpuDriverWindows.enabled) {
+  name: '${uniqueString(deployment().name, location)}-VM-NvidiaGpuDriverWindows'
+  params: {
+    virtualMachineName: vm.name
+    name: 'NvidiaGpuDriverWindows'
+    location: location
+    publisher: 'Microsoft.HpcCompute'
+    type: 'NvidiaGpuDriverWindows'
+    typeHandlerVersion: contains(extensionNvidiaGpuDriverWindows, 'typeHandlerVersion') 
+     ? extensionNvidiaGpuDriverWindows.typeHandlerVersion 
+     : '1.4'
+    autoUpgradeMinorVersion: contains(extensionNvidiaGpuDriverWindows, 'autoUpgradeMinorVersion') 
+     ? extensionNvidiaGpuDriverWindows.autoUpgradeMinorVersion 
+     : true
+    enableAutomaticUpgrade: contains(extensionNvidiaGpuDriverWindows, 'enableAutomaticUpgrade') 
+     ? extensionNvidiaGpuDriverWindows.enableAutomaticUpgrade 
+     : false
+    tags: extensionNvidiaGpuDriverWindows.?tags ?? tags
+  }
+  dependsOn: [
+    vm_azureDiskEncryptionExtension
+  ]
+}
+
+module vm_hostPoolRegistrationExtension 'extension/main.bicep' = if (extensionHostPoolRegistration.enabled) {
+  name: '${uniqueString(deployment().name, location)}-VM-HostPoolRegistration'
+  params: {
+    virtualMachineName: vm.name
+    name: 'HostPoolRegistration'
+    location: location
+    publisher: 'Microsoft.PowerShell'
+    type: 'DSC'
+    typeHandlerVersion: contains(extensionHostPoolRegistration, 'typeHandlerVersion') 
+     ? extensionHostPoolRegistration.typeHandlerVersion 
+     : '2.77'
+    autoUpgradeMinorVersion: contains(extensionHostPoolRegistration, 'autoUpgradeMinorVersion') 
+     ? extensionHostPoolRegistration.autoUpgradeMinorVersion 
+     : true
+    enableAutomaticUpgrade: contains(extensionHostPoolRegistration, 'enableAutomaticUpgrade') 
+     ? extensionHostPoolRegistration.enableAutomaticUpgrade 
+     : false
+    settings: {
+      modulesUrl: 'https://wvdportalstorageblob.blob.${environment().suffixes.storage}/galleryartifacts/Configuration_09-08-2022.zip'
```

Comment:
I wonder if that value should be configurable. It would appear the file name could change over time? Even the storage account has technically an outdated name (wvd vs avd).
Same question for the configuration function

Reply:
Moved it also into the host pool registration object

Reply:
What about the aforementioned configuration function?

Reply:
Now it needs to be defined by the consumer of the module. In the test they can see how it works and maybe change the URL, if needed

Reply:
But isn't it still hardcoded? I can see the following in the template:
```bicep
configurationFunction: 'Configuration.ps1\\AddSessionHost'
```

I'd suggest to update to something like 
```bicep
configurationFunction: extensionHostPoolRegistration.?configurationFunction ?? 'Configuration.ps1\\AddSessionHost'
```

Reply:
done